src/App.js

```diff
@@ -1,11 +1,61 @@
-import Calculator from './components/Calculator';
+import { useEffect, useState } from 'react';
+import { API_URL, HEADERS_LIST } from './api/apiSettings';
+import Calculator from './components/Calculator/Calculator';
+import Quote from './components/Quote/Quote';
 import './assets/styles/App.css';
 
-const App = () => (
-  <>
-    <h1>Welcome to Math Magicians</h1>
-    <Calculator />
-  </>
-);
+const App = () => {
+  const [dailyQuote, setDailyQuote] = useState({ quote: '', author: '' });
+  const [isLoading, setIsLoading] = useState(false);
+  const [hasError, setHasError] = useState(false);
+
+  useEffect(() => {
+    const fetchQuote = async (quoteType = 'inspirational') => {
+      setIsLoading(true);
+
+      const fetchUrl = `${API_URL}${quoteType}`;
+      const request = new Request(
+        fetchUrl, {
+          method: 'GET',
+          'Content-Type': 'application/json',
+          headers: HEADERS_LIST,
+        },
+      );
+
+      try {
+        const response = await fetch(request);
+        const [json] = await response.json();
+        setDailyQuote((prev) => ({
+          ...prev,
+          ...json,
+        }));
+
+        setHasError(false);
+      } catch (error) {
+        setHasError(true);
+      }
+
+      setIsLoading(false);
+    };
+
+    fetchQuote();
+  }, []);
```

Comment:
- [ ] Great job handling errors very well. :+1: Let's do a bit of refactoring to achieve the project requirements. All these data fetching and error handling belongs to the `Quote` component, we are not using any of this data in the `App` component itself or any of its children other than `Quote`. Kindly move this logic to the `Quote` component. 🤝️

**Project requirements**
![Screenshot from 2023-03-23 05-09-11](https://user-images.githubusercontent.com/17243325/227082595-ab79721b-432d-4af2-a039-591946bd9b59.png)

Reply:
Thanks for your feedback @amejid.
All Quote logic was move from App into Quote component.